src/metabase/query_analysis.clj

```diff
@@ -91,8 +91,8 @@
 
 (defn- explicit-references [field-ids]
   (let [field-refs (explicit-field-references field-ids)]
-    {:fields field-refs
-     :tables (distinct (map #(dissoc % :field-id :field) field-refs))}))
+    {:fields (distinct field-refs)
```

Comment:
This one is for good luck, the analyzer's *shouldn't* send back duplicates - but maybe they can.